packages/dds/tree/src/test/simple-tree/benchmarkUtilities.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import assert from "assert";
+import {
+	SchemaFactory,
+	type NodeFromSchema,
+	type ValidateRecursiveSchema,
+} from "../../simple-tree/index.js";
+import { hydrate } from "./utils.js";
+import { Tree } from "../../shared-tree/index.js";
+
+const schemaFactory = new SchemaFactory("test");
+
+/**
+ * A recursive tree node with a single leaf value
+ */
+export class DeepTreeNode extends schemaFactory.objectRecursive("deep-tree-node", {
+	node: [() => DeepTreeNode, schemaFactory.number],
+}) {}
+{
+	type _check = ValidateRecursiveSchema<typeof DeepTreeNode>;
+}
+
+/**
+ * A wide tree node with all leaves having the same value
+ */
+export const WideTreeNode = schemaFactory.array(schemaFactory.number);
+export type WideTreeNode = NodeFromSchema<typeof WideTreeNode>;
+
+/**
+ * Make a deep tree with a single leaf value
+ */
+function makeDeepTree(depth: number, leafValue: number): DeepTreeNode {
+	return depth === 1
+		? new DeepTreeNode({ node: leafValue })
+		: new DeepTreeNode({ node: makeDeepTree(depth - 1, leafValue) });
+}
+
+/**
+ * generate a deep tree with a single leaf value
+ */
+export function generateDeepSimpleTree(depth: number, leafValue: number): DeepTreeNode {
+	assert(Number.isSafeInteger(depth));
+	assert(depth > 0);
+
+	return hydrate(DeepTreeNode, makeDeepTree(depth, leafValue));
+}
+
+/**
+ * read the deep tree and return the depth and value of the leaf node
+ */
+export function readDeepSimpleTree(tree: DeepTreeNode): {
+	depth: number;
+	value: number;
+} {
+	let currentNode: DeepTreeNode | number = tree;
+	let depth = 0;
+
+	while (Tree.is(currentNode, DeepTreeNode)) {
+		depth += 1;
+		currentNode = currentNode.node;
+	}
+
+	const value = currentNode ?? assert.fail("missing leaf value");
+	return { depth, value };
+}
+
+/**
+ * Update the deep tree with a new value
+ */
+export function writeDeepSimpleTree(tree: DeepTreeNode, newValue: number): void {
+	let currentNode: DeepTreeNode | number = tree;
+
+	while (!Tree.is(currentNode.node, schemaFactory.number)) {
+		currentNode = currentNode.node;
+	}
+
+	currentNode.node = newValue;
+}
+
+/**
+ * generate a wide tree with a single layer
+ */
+export function generateWideSimpleTree(width: number, leafValue: number): WideTreeNode {
+	const values = [];
+	for (let i = 0; i < width; i++) {
+		values.push(leafValue);
+	}
+	return hydrate(WideTreeNode, values);
+}
```

Comment:
We can make this simpler and more declarative using Array.from:
```suggestion
export function generateWideSimpleTree(length: number, leafValue: number): WideTreeNode {
	return hydrate(WideTreeNode, Array.from({ length }, () => leafValue));
}
```

Reply:
Updated